Arduino/startUpwithBluetooth/startUpwithBluetooth.ino

```diff
@@ -0,0 +1,70 @@
+#include <Smartcar.h>
+
+const unsigned short LEFT_ODOMETER_PIN = 2;
+const unsigned short RIGHT_ODOMETER_PIN = 3;
+const unsigned long PRINT_INTERVAL = 100;
+unsigned long previousPrintout = 0;
+const int TRIGGER_PIN = 6; //D6
+const int ECHO_PIN = 7; //D7
+const unsigned int MAX_DISTANCE = 100;
+char blue;
+SR04 front(TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);                //bluetooth value reciever
+
+
+BrushedMotor leftMotor(8, 10, 9);
+BrushedMotor rightMotor(12, 13, 11);
+DifferentialControl control(leftMotor, rightMotor);
+
+GY50 gyroscope(37);
+DirectionlessOdometer leftOdometer(100);
+DirectionlessOdometer rightOdometer(100);
+
+SmartCar car(control, gyroscope, leftOdometer, rightOdometer);
+
+void setup() {
+  Serial.begin(9600);
+  // Initialize the odometers (they won't work otherwise)
+  leftOdometer.attach(LEFT_ODOMETER_PIN, []() {
+    leftOdometer.update();
+  });
+  rightOdometer.attach(RIGHT_ODOMETER_PIN, []() {
+    rightOdometer.update();
+  });
+
+  car.enableCruiseControl();
+ // car.setSpeed(1.5);
+}
+
+void loop() {
+  // Maintain the speed and update the heading
+    car.update();
+   
+    while (Serial.available()) blue = Serial.read(); 
```

Comment:
This is very tricky... If there are more than one incoming characters `blue` will contain the last of them and discard all the previous ones. Is that really what you intend?

Reply:
we are aware about this problem but we haven't found the right solution for this problem yet.